snooty/eventparser.py

```diff
@@ -30,16 +58,16 @@ def get_event_listeners(self, event: str) -> Set[Callable[..., Any]]:
     def fire(
         self,
         event: str,
-        filename: FileId,
+        fileid: Union[FileId, FileIdStack],
```

Comment:
Do you see any benefit to only using a `FileIdStack` (no union) here? It seems like the page enter/exit events could work with a stack of length 1. But maybe I am missing something!

Reply:
No real strong opinion, so I just went ahead and changed the PR to use FileIdStack everywhere!